### DISCUSSION TITLE
chess engine tested run smoothly with python-chess

### DISCUSSION THREAD

**Discussion by sugizo** [niklasf/python-chess#877] [General]
**mac** 
- defenchess
- fairy_stockfish
- fruit
- glaurung
- komodo
- sjaakii
- stockfish
- toga2
- winter
- wyldchess

**debian/ubuntu base repo, install via apt (tested on google colab)**
- fruit
- glaurung
- sjaakii
- stockfish
- toga2

tested another engines too but, some of them throw an error during compiled or during running with python-chess

anyone can share which chess engine tested run smoothly with python-chess ?

best regards

**Comment by niklasf** [comment]
Thanks for the overview!

If any of the cases where communication fails look like python-chess is to blame, please open bug reports and I'll look into it.

**Comment by sugizo** [comment]
here's the steps on google colab (ubuntu bionic), the success one
```
if 'google.colab' in str(get_ipython() ):
    !apt-get install -y fruit glaurung sjaakii stockfish toga2
    fruit_path = "/usr/games/fruit"
    glaurung_path = "/usr/games/glaurung"
    sjaakii_path = "/usr/games/sjaakii"
    stockfish_path = "/usr/games/stockfish"
    toga2_path = "/usr/games/toga2"
```
then put the path above (whatever engine you want to use) into python-chess 
e.g.
`engine = chess.engine.SimpleEngine.popen_uci(stockfish_path)
`

then if you want to try another chess engine on google colab or ubuntu/debian base you can search it
`!apt search chess`
after got the list then install it, 
i tried it all, but the 5 mentions above is succeed (note tested on ubuntu bionic repo google colab, not sure with another repo)

and for engine with more effort, compiled from source (github, gitlab, etc), only tested it on mac

**Comment by sugizo** [comment]
tested work on ubuntu and google colab for analysing and evaluating from pgn file
**xboard**
![image](https://user-images.githubusercontent.com/2418295/234747010-3ddf94fd-21b5-4300-a8aa-ae1684c363bf.png)

**uci**
![image](https://user-images.githubusercontent.com/2418295/234747046-83b85458-6451-4ada-8048-7cc44903fb0b.png)

**uci combine with stockfish (python module pypi)**
![image](https://user-images.githubusercontent.com/2418295/234747107-b4e6975c-9b9d-4883-bd05-1d9bac0b82bc.png)
